src/app/shared/directives/tooltip.directive.ts

```diff
@@ -0,0 +1,63 @@
+import { ApplicationRef, ComponentRef, createComponent, Directive, EventEmitter, HostListener, Input, OnDestroy, Output, TemplateRef } from '@angular/core'
+import { TooltipComponent } from '../components/tooltip/tooltip.component'
+
+export interface TooltipMetadata {
+  xPosition: number
+  width?: number
+}
+
+@Directive({
+  selector: '[sofieTooltip]',
+})
+export class TooltipDirective implements OnDestroy {
+  @Input() public sofieTooltip: TemplateRef<unknown>
+  @Input() public shouldFollowCursor: boolean = false
+
+  @Output() public tooltipMetadata: EventEmitter<TooltipMetadata> = new EventEmitter()
+
+  private tooltipComponentRef: ComponentRef<TooltipComponent>
+
+  constructor(private readonly applicationRef: ApplicationRef) {}
+
+  @HostListener('mouseenter', ['$event'])
+  public showTooltip(event: MouseEvent): void {
+    this.createTooltip()
+    this.moveTooltip(event)
+  }
+
+  private createTooltip(): void {
+    this.tooltipComponentRef = createComponent(TooltipComponent, {
+      environmentInjector: this.applicationRef.injector,
+    })
+
+    this.tooltipComponentRef.instance.templateRef = this.sofieTooltip
+    this.tooltipComponentRef.changeDetectorRef.detectChanges()
+
+    document.body.append(this.tooltipComponentRef.location.nativeElement)
+  }
+
+  @HostListener('mousemove', ['$event'])
+  public moveTooltip(event: MouseEvent): void {
+    if (!this.tooltipComponentRef) {
+      return
+    }
+
+    const top: number = event.clientY - event.offsetY
+    const tooltipWidth = this.tooltipComponentRef.instance.tooltipContainer.nativeElement.offsetWidth
+    const left: number = Math.min(Math.max(event.clientX - Math.ceil(tooltipWidth / 2), 0), window.innerWidth - tooltipWidth)
```

Comment:
A fair bit of math is happening here. To make it understandable for us in the future, we could extract some of the parts out and name them. A suggestion:
```typescript
const MIN_HORIZONTAL_OFFSET_IN_PIXELS: number = 0
// ...
const horizontalOffsetInPixels: number = event.clientX - Math.ceil(tooltipWidth / 2)
const maxHorizontalOffsetInPixels: number = window.innerWidth - tooltipWidth
const constrainedHorizontalOffsetInPixels = Math.min(
  Math.max(horizontalOffsetInPixels, MIN_HORIZONTAL_OFFSET_IN_PIXELS),
  maxHorizontalOffsetInPixels
)

Reply:
Updated. Made some slight changes to your suggestion.